src/main/java/com/metamx/common/StringUtils.java

```diff
@@ -66,4 +67,21 @@ public static String fromUtf8(final ByteBuffer buffer)
       throw Throwables.propagate(e);
     }
   }
+
+  public static String safeFormat(String message, Object... formatArgs)
+  {
+    if(formatArgs == null || formatArgs.length == 0) {
+      return message;
+    }
+    try {
+      return String.format(message, formatArgs);
```

Comment:
could still throw NPE, but that's probably fine as that is a really strange error for a function that usually has literals passed into it.